test/widgets/message_list_test.dart

```diff
@@ -51,6 +59,58 @@ Future<void> setupMessageListPage(WidgetTester tester, {
 void main() {
   TestZulipBinding.ensureInitialized();
 
+  group('MessageWithSender', () {
+    testWidgets('Updates avatar on RealmUserUpdateEvent', (tester) async {
+      addTearDown(TestZulipBinding.instance.reset);
+
+      RealmContentNetworkImage? findAvatarImageWidget(WidgetTester tester) {
+        final firstMessageWithSender = tester.widgetList(find.byType(MessageWithSender)).first;
+        return tester.widgetList<RealmContentNetworkImage>(
+          find.descendant(
+            of: find.byWidget(firstMessageWithSender),
+            matching: find.byType(RealmContentNetworkImage)),
+        ).singleOrNull;
+      }
+
+      void checkResultForSender(User sender) {
+        final avatarUrl = sender.avatarUrl;
+        switch (avatarUrl) {
+          case String(): {
```

Comment:
Also when the switch is just "null or not" it's probably simplest to just use an if/else :slightly_smiling_face: